ios/brave-ios/Sources/Brave/Frontend/Browser/New Tab Page/Backgrounds/NewTabPageVideoPlayer.swift

```diff
@@ -135,6 +137,33 @@ class NewTabPageVideoPlayer {
     }
   }
 
+  private func stopFrameTime() -> CMTime {
+    var stopFrameTime = CMTime(
+      seconds: kMaxAutoplayDuration,
+      preferredTimescale: CMTimeScale(NSEC_PER_SEC)
+    )
+
+    let stopFrameNumber = parseStopFrameNumberFromFilename(
+      filename: backgroundVideoPath.lastPathComponent
+    )
+    if let stopFrameNumber, let frameRate {
+      // Calculate the time to stop the video at the specified frame.
+      let stopFrameTimeInSeconds = (Double(stopFrameNumber) + 0.5) / frameRate
```

Comment:
We could return a Double from parseStopFrameFromFilename rather than casting

Reply:
Fixed